x-pack/plugins/security_solution/docs/prebuilt_rules_customization_rfc.md

```diff
@@ -0,0 +1,2861 @@
+
+All of the fields in the UI, listed above, are currently editable for custom rules, except for rule type, which is read only.
+
+Once done editing the rule, the user clicks on the "Save Changes" button, which calls the **Update Rule** - `PUT /rules` endpoint, passing the payload for the whole rule.
+
+**Expected behaviour for customizing prebuilt rules**
+
+- All four tabs of the Rule Edit page should be enabled, and all the fields within each tab should be editable, as they currently are for custom rules.
+- The only field in the UI that should not be customizable is **Rule Type**, that should continue to be read-only.
+- **Definition** should be the default open tab when opening the edit rule page for a prebuilt rule (current default is **Actions**)
+- Field validation should continue to work as it does for custom rules.
+- No fields should return a validation error for the values that come from the `security_detection_engine` package prebuilt rules. This means that a user should be able to successfully save the prebuilt rule with no changes. See **List of issues to fix** below.
+
+#### Via Bulk Actions
+
+Custom rules can currently be updated via the Rule Table's **Bulk Actions**, which uses the **Bulk Actions** - `POST /rules/_bulk_action` endpoint.
+
+Apart from enabling/disabling a rule - the only other action that modifies a rule's saved object and is currently possible for prebuilt rules - the user can use bulk actions to:
+
+- Add, delete and overwrite **index patterns**
+- Add, delete and overwrite **tags**
+- Add and overwrite **rule actions**
+- Update the rule's **schedule: interval and lookback time**
+- Select a **timeline template** for the rule
+
+As explained above, the UI validates that the five actions listed above are only possible to custom rules by using the `dryRun` mode of the **bulk actions** endpoint. If any of the selected rules in the request payload is a prebuilt rule, a message is displayed to the user informing them that modifying prebuilt rules is not possible, and that rule is removed from the subsequent request that is made in normal mode to actually edit the rule.
+
+**Expected behaviour for customizing prebuilt rules**
+
+- The Bulk Actions UI should now allow users to edit prebuilt rules.
+- All bulk actions actions listed above should now be applicable to prebuilt rules (This means that the changes in the API for the validation done in `dryRun` mode should not remove the rules from the payload anymore).
+
+> The `dryRun` mode request will still be necessary since we perform other checks that can still prevent prebuilt rules from being edited. For example, Machine Learning and ES|QL rules cannot be bulk edited to add or delete index patterns, since they don't have that field in their schema.
+
+#### Via the Rules Details Page
+
+The Rules Details page allows the user to modify the rule in two ways:
```

Comment:
Can we also describe in the RFC what changes to the UI we will need to introduce for the Rule Details and Rule Management pages?

Reply:
Added